crates/interpreter/src/opcode.rs

```diff
@@ -276,43 +284,140 @@ impl OpCode {
 }
 
 /// Information about opcode, such as name, and stack inputs and outputs.
-#[derive(Clone, Copy, Default, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct OpCodeInfo {
-    pub name: &'static str,
-    pub inputs: u8,
-    pub outputs: u8,
-    // TODO make this a bitfield
-    pub is_eof: bool,
-    // If the opcode is return from execution. aka STOP,RETURN, ..
-    pub is_terminating_opcode: bool,
-    /// Size of opcode with its intermediate bytes.
+    /// Invariant: `(name_ptr, name_len)` is a `&'static str`. It is a shorted variant of `str` as the name size is always 
+    /// less than 256 characters.
```

Comment:
```suggestion
    /// Invariant: `(name_ptr, name_len)` is a `&'static str`. It is a shorted variant of `str` as
    /// the name length is always less than 256 characters.
```

not sure why fmt is failing here, checking doc length